backend/vehicle/views.py

```diff
@@ -76,3 +66,27 @@ def get(self, request, vehicle_id, *args, **kwargs):
             # return Response(serialized_data.data, status=status.HTTP_200_OK)
         except vehicle.DoesNotExist:
             return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+class VehicleFilterList(APIView):
+    """
+    Get list of vehicles based off of filter
+    Takes limit + offset from url
+    """
+
+    def get_queryset(self):
+        queryset = infinite_filter(self.request)
+        return queryset
+
+    def get(self, request):
+        url_parameter = request.GET.get("p")
+        if url_parameter:
+            vehicles = self.get_queryset()
+
+            serialized_data = VehicleSerializer(vehicles, many=True)
+
+            return Response(
+                {"vehicles": serialized_data.data, "more_data": has_more_data(request)}
+            )
+
+        return Response(VehicleSerializer(Vehicle.objects.all()[:10], many=True).data)
```

Comment:
@kevinrczhang Is there a way to merge filtering with the current GET query for `VehicleListApiView`? Where if we pass in a page then we use the pagination, but, if not, we get an infinite list. Also, could we rename the name of the required variables in the request from `p`, `l`, and `o` to  `page`, `limit`, and `offset` respectively.

Reply:
changed "p" to "search"